### PR TITLE
A small step towards removing C-style loops

### PR DESCRIPTION
**Pull request by hamin**
This is a small step towards removing C-style loops in favor of Swift for/in loops as presented in this accepted proposal:https://github.com/apple/swift-evolution/pull/23/commits . Nothing big here, just getting my feet wet and trying help improve code quality :)

### CODE REVIEW

**Comment by gribozavr** [comment]
I'll run tests and merge.

File: stdlib/public/core/Prespecialized.swift [hunk 1]
```diff
@@ -29,8 +29,8 @@ struct _Prespecialize {
         a[0] = a[j]
       }
 
-      for var i1 = 0; i1 < a.count; ++i1 {
-        for var i2 = 0; i2 < a.count; ++i2 {
+      for i1 in 0..<a.count {
```
Comment (dabrahams) [inline stdlib/public/core/Prespecialized.swift]: Shouldn't this use a.indices?

**Comment by gribozavr** [comment]
@dabrahams Just noticed your comment.  I'll take care of that.

**Comment by hamin** [comment]
@gribozavr I don't mean to sound cheesy, but this is the first commit I've ever made to a programming language ever. Thank you :smile:

**Comment by gribozavr** [comment]
@hamin You're welcome! :100:

**Pull request merged**